pkg/splunk/test/util.go

```diff
@@ -42,3 +49,107 @@ func GetMockS3SecretKeys(name string) corev1.Secret {
+	}
+	client.MockReturnContexts[cmd] = mockPodExecReturnContext
+}
+
+// AddMockPodExecReturnContexts adds mockPodExecReturnContexts for the corresponding commands
+func (client *MockPodExecClient) AddMockPodExecReturnContexts(podExecCmds []string, mockPodExecReturnContexts ...*MockPodExecReturnContext) {
+	for n := range mockPodExecReturnContexts {
+		client.AddMockPodExecReturnContext(podExecCmds[n], mockPodExecReturnContexts[n])
+	}
+}
+
+// GetMockPodExecReturnContextAndKey returns the mockPodExecReturnContext object and the corresponding command
+func (client *MockPodExecClient) GetMockPodExecReturnContextAndKey(cmd string) (*MockPodExecReturnContext, string) {
+	for key := range client.MockReturnContexts {
+		if strings.Contains(cmd, key) {
+			return client.MockReturnContexts[key], key
+		}
+	}
+	return nil, ""
+}
+
+// CheckPodExecCommands method for MockPodExecClient checks if got commands are same as received commands
+func (client *MockPodExecClient) CheckPodExecCommands(t *testing.T, testMethod string) {
+	if len(client.GotCmdList) != len(client.WantCmdList) {
+		t.Fatalf("%s got %d number of commands; want %d number of commands", testMethod, len(client.GotCmdList), len(client.WantCmdList))
+	}
+	for n := range client.GotCmdList {
+		if client.GotCmdList[n] != client.WantCmdList[n] {
+			t.Errorf("%s GotCmdList[%d]=%s, want %s;", testMethod, n, client.GotCmdList[n], client.WantCmdList[n])
+		}
+	}
+}
+
+// RunPodExecCommand returns the dummy values for mockPodExecClient
+func (client *MockPodExecClient) RunPodExecCommand(streamOptions *remotecommand.StreamOptions, baseCmd []string) (string, string, error) {
+
+	var mockPodExecReturnContext *MockPodExecReturnContext = &MockPodExecReturnContext{}
+	var command string
+	// This is to prevent the crash in the case where streamOptions.Stdin is anything other than *strings.Reader
+	// In most of the cases the base command will be /bin/sh but if it is something else, it can be reading from
+	// a io.Reader pipe. For e.g. tarring a file, writing to a write pipe and then untarring it on the pod by reading
+	// from the reader pipe.
+	if baseCmd[0] == "/bin/sh" {
+		var cmdStr string
+		streamOptionsCmd := streamOptions.Stdin.(*strings.Reader)
+		for i := 0; i < int(streamOptionsCmd.Size()); i++ {
+			cmd, _, _ := streamOptionsCmd.ReadRune()
+			cmdStr = cmdStr + string(cmd)
+		}
+
+		mockPodExecReturnContext, command = client.GetMockPodExecReturnContextAndKey(cmdStr)
+		if mockPodExecReturnContext == nil {
+			err := fmt.Errorf("mockPodExecReturnContext is nil")
+			return "", "", err
+		}
+	}
+
+	// check if the command is already added or not in the list of GotCmdList
+	var found bool
+	for i := range client.GotCmdList {
+		if reflect.DeepEqual(client.GotCmdList[i], command) {
+			found = true
+			break
+		}
+	}
+	if !found {
+		client.GotCmdList = append(client.GotCmdList, command)
+	}
+
+	return mockPodExecReturnContext.StdOut, mockPodExecReturnContext.StdErr, mockPodExecReturnContext.Err
```

Comment:
Does this mock logic works only if the base command is `"/bin/sh"`?

Reply:
The logic from line 118-129 works only for command "bin/sh" at the moment and the reason is explained in the comment section. Everything else is common.